simpleclient_dropwizard/src/main/java/io/prometheus/client/dropwizard/DropwizardExports.java

```diff
@@ -9,37 +11,49 @@
 import java.util.concurrent.TimeUnit;
 import java.util.logging.Level;
 import java.util.logging.Logger;
-import java.util.regex.Pattern;
 
 /**
  * Collect Dropwizard metrics from a MetricRegistry.
  */
 public class DropwizardExports extends io.prometheus.client.Collector implements io.prometheus.client.Collector.Describable {
-    private MetricRegistry registry;
     private static final Logger LOGGER = Logger.getLogger(DropwizardExports.class.getName());
+    private MetricRegistry registry;
+    private SampleBuilder sampleBuilder;
 
     /**
+     * Creates a new DropwizardExports with a {@link DefaultSampleBuilder}.
+     *
      * @param registry a metric registry to export in prometheus.
      */
     public DropwizardExports(MetricRegistry registry) {
         this.registry = registry;
+        this.sampleBuilder = new DefaultSampleBuilder();
+    }
+
+    /**
+     * @param registry      a metric registry to export in prometheus.
+     * @param sampleBuilder sampleBuilder to use to create prometheus samples.
+     */
+    public DropwizardExports(MetricRegistry registry, SampleBuilder sampleBuilder) {
+        this.registry = registry;
+        this.sampleBuilder = sampleBuilder;
+    }
+
+    private static String getHelpMessage(String metricName, Metric metric) {
+        return String.format("Generated from Dropwizard metric import (metric=%s, type=%s)",
+                metricName, metric.getClass().getName());
     }
 
     /**
      * Export counter as Prometheus <a href="https://prometheus.io/docs/concepts/metric_types/#gauge">Gauge</a>.
      */
     List<MetricFamilySamples> fromCounter(String dropwizardName, Counter counter) {
         String name = sanitizeMetricName(dropwizardName);
-        MetricFamilySamples.Sample sample = new MetricFamilySamples.Sample(name, new ArrayList<String>(), new ArrayList<String>(),
+        MetricFamilySamples.Sample sample = sampleBuilder.createSample(dropwizardName, "", new ArrayList<String>(), new ArrayList<String>(),
                 new Long(counter.getCount()).doubleValue());
         return Arrays.asList(new MetricFamilySamples(name, Type.GAUGE, getHelpMessage(dropwizardName, counter), Arrays.asList(sample)));
```

Comment:
Is the metric name going to match the timeseries name?

Reply:
Changed so they match